### DISCUSSION TITLE
Question about calculating similarity score

### DISCUSSION THREAD

**Discussion by amarsh26** [antismash/antismash#637] [Q&A] [answered]
Hi, 

My research group is working on adding antiSMASH to KBase, and I am writing a script to calculate similarity score from the json output to store in a tsv for later use. To do this, I am using the directions supplied in #608 ("To recalculate the similarity, divide hits from the breakdown (the second element) by the total number of genes in the reference (proteins in the first element)". However, when testing my script, I used the example data supplied (NC_0003888.3) and noticed a few differences between what I calculated from the json and what is supplied in the html. For example, region 22 has a similarity score of 96% and I calculated 104%. In the json, the number hits were 26 and the number of proteins were 25. I was wondering if there is another way you would recommend calculating the score. It also seems that similarity score is always rounded down, and I wanted to confirm. 

Thanks so much for your help!

**Comment by SJShaw** [comment, marked as answer]
That region has 26 total matches for genes in the reference, but only 24 of them are unique (i.e. two of the hits match reference genes that already have hits from other query genes). That's why it's 96%. Any similarity over 100% should raise alarms, since it's calculated with `number of reference genes with hits / number of reference genes`.

The floor is always taken of the final percentage. More than two digits of precision doesn't add much when it's extremely unlikely for the reference clusters have more than 100 genes.

**Reply by amarsh26** [reply]
Thanks so much for your clarification!
